lsl/correlator/uvutils.py

```diff
@@ -156,117 +158,70 @@ def compute_uvw(antennas, HA=0.0, dec=34.070, freq=49.0e6, site=lwa1, include_au
+        HA2 = HA * 15.0 * np.pi/180
+    if isinstance(dec, astro.dms):
+        dec2 = dec.to_rad()
     elif isinstance(dec, AstroAngle):
-        dec2 = dec.radian
+        dec2 = dec.rad
+    elif isinstance(dec, ephem.Angle):
+        dec2 = dec*1.0
     else:
-        dec2 = dec * numpy.pi/180
-    if isinstance(site, AstroEarthLocation):
-        lat2 = site.lat.radian
+        dec2 = dec * np.pi/180
+        
+    # Setup - Target and antennas in ITRF
+    try:
+        t = AstroTime(site.date + astro.DJD_OFFSET, format='jd', scale='utc')
+    except AttributeError:
+        t = AstroTime.now()
+    if isinstance(site, EarthLocation):
+        el = site
     else:
-        lat2 = site.lat
+        el = site.earth_location
+    ha = HADec(HA2*astrounits.rad, dec2*astrounits.rad, location=el, obstime=t)
+    ants = [ant.stand.earth_location for ant in antennas]
+    ants = EarthLocation(x=astrounits.Quantity([ant.x for ant in ants]),
+                         y=astrounits.Quantity([ant.y for ant in ants]),
+                         z=astrounits.Quantity([ant.z for ant in ants]))
+    
     
     # Coordinate transformation matrices
-    trans1 = numpy.matrix([[0, -numpy.sin(lat2), numpy.cos(lat2)],
-                           [1,  0,               0],
-                           [0,  numpy.cos(lat2), numpy.sin(lat2)]])
-    trans2 = numpy.matrix([[ numpy.sin(HA2),                  numpy.cos(HA2),                 0],
-                           [-numpy.sin(dec2)*numpy.cos(HA2),  numpy.sin(dec2)*numpy.sin(HA2), numpy.cos(dec2)],
-                           [ numpy.cos(dec2)*numpy.cos(HA2), -numpy.cos(dec2)*numpy.sin(HA2), numpy.sin(dec2)]])
-                    
+    site_p, site_v = el.get_gcrs_posvel(t)
+    antpos_c_ap = GCRS(ants.get_gcrs_posvel(t)[0], 
+                       obstime=t, obsgeoloc=site_p, obsgeovel=site_v)
+    frame_uvw = ha.transform_to(antpos_c_ap)
+    uvw_transform = rotation_matrix(0*astrounits.deg, 'x') \
+                     @ rotation_matrix(-frame_uvw.spherical.lat, 'y') \
+                     @ rotation_matrix(frame_uvw.spherical.lon, 'z')
+    
+    # Transform antennas into the (w,u,v) frame
+    bl_ants = []
+    for ant in antpos_c_ap:
+        c = ant.cartesian
+        bl_ants.append(uvw_transform @ np.array([c.x.to('m').value, c.y.to('m').value, c.z.to('m').value]))
+        
+    # Compute the baselines and convert to wavelengths
     for k,(i,j) in enumerate(baselines):
-        # Go from a east, north, up coordinate system to a celestial equation, 
-        # east, north celestial pole system
-        xyzPrime = antennas[i].stand - antennas[j].stand
-        xyz = trans1*numpy.matrix([[xyzPrime[0]], [xyzPrime[1]], [xyzPrime[2]]])
+        bl = (bl_ants[i] - bl_ants[j])[[1,2,0]]
+        bl.shape += (1,)
         
-        # Go from CE, east, NCP to u, v, w
-        temp = trans2*xyz
-        uvw[k,:,:] = temp[:,0] * freq.ravel() / speedOfLight
+        uvw[k,:,:] = bl * freq.ravel() / speedOfLight
```

Comment:
I don't really like how this is implemented.  I think the old method was easier to understand.  I also think there might be an astropy version problem with `HADec`.

Reply:
Reverted.